### PR TITLE
Trace test

### PR DESCRIPTION
**Pull request by dilner**

### CODE REVIEW

File: tests/on_target/tests/test_traces.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    logger.info("Dumping modem traces to uart1")
+    uart1_before_dump = t91x_traces.trace.get_size()
+    t91x_traces.uart.write("modem_trace dump_uart\r\n")
+    t91x_traces.uart.wait_for_str_ordered(["Reading out", "bytes of trace data"], timeout=5)
+    modem_trace_size_to_read = t91x_traces.uart.extract_value(r"Reading out (\d+) bytes of trace data")
+    assert modem_trace_size_to_read
+    assert int(modem_trace_size_to_read[0]) >= int(modem_trace_size_2[0])
+    start_t = time.time()
+    while True:
+            uart1_size = t91x_traces.trace.get_size()
```
Comment (jorgenmk) [inline tests/on_target/tests/test_traces.py:55]: Double indentation here

File: tests/on_target/tests/test_traces.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    assert modem_trace_size_2
+    assert int(modem_trace_size_2[0]) > int(modem_trace_size_1[0])
+
+    logger.info("Dumping modem traces to uart1")
+    uart1_before_dump = t91x_traces.trace.get_size()
+    t91x_traces.uart.write("modem_trace dump_uart\r\n")
+    t91x_traces.uart.wait_for_str_ordered(["Reading out", "bytes of trace data"], timeout=5)
+    modem_trace_size_to_read = t91x_traces.uart.extract_value(r"Reading out (\d+) bytes of trace data")
+    assert modem_trace_size_to_read
+    assert int(modem_trace_size_to_read[0]) >= int(modem_trace_size_2[0])
```
Comment (jorgenmk) [inline tests/on_target/tests/test_traces.py:52]: Why do we use `>=` operator here? I would expect `==`?

File: tests/on_target/utils/test_uart.py [hunk 3]
```diff
@@ -105,3 +105,54 @@ def test_wait_ordered_7_none(time_sleep, time_time):
     with pytest.raises(AssertionError) as ex_info:
         u.wait_for_str_ordered(["abc", "def", "ghi", "jkl"], timeout=2)
     assert "abc missing" in str(ex_info.value)
+
+@patch("time.time", side_effect=counter())
+@patch("time.sleep")
+def test_wait_8_get_current_size(time_sleep, time_time):
+    u = mocked_uart()
+    u.log = "foo123\nbar123\nbaz123\n"
+    current_log_size = u.wait_for_str(["bar"], timeout=3)
```
Comment (jorgenmk) [inline tests/on_target/utils/test_uart.py:114]: Did you intend to run `u.get_size()` here?

Comment (dilner) [reply tests/on_target/utils/test_uart.py:114]: I intended to check that u.wait_for_str returns the log size. But maybe we should have a test for get_size as well?

**Pull request merged**